modules/pricer/src/main/java/com/opengamma/strata/pricer/DiscountFactors.java

```diff
@@ -120,6 +120,17 @@ public default double discountFactor(LocalDate date) {
    * @throws RuntimeException if the value cannot be obtained
    */
   public abstract double discountFactor(double yearFraction);
+  
+  /**
+   * Returns the discount factor derivative with respect to the year fraction or time.
+   * <p>
+   * The year fraction must be based on {@code #relativeYearFraction(LocalDate)}.
+   * 
+   * @param yearFraction  the year fraction 
+   * @return the discount factor derivative
+   * @throws RuntimeException if the value cannot be obtained
+   */
+  public abstract double discountFactorTimeDerivative(double yearFraction);
```

Comment:
This is a backwards incompatible change. It is probably the right thing to do, but will need noting in release notes.